### PR TITLE
[API Compatibility] paddle.sigmoid, paddle.nn.functional.sigmoid sink into C++

### PR DESCRIPTION
**Pull request by hushenwei2000**
### PR Category
Operator Mechanism

### PR Types
New features

### Description
下沉paddle.sigmoid 到C++层，支持Torch的传参方法。

pcard-67164

### CODE REVIEW

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: test/legacy_test/test_sigmoid.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zyfncg) [inline test/legacy_test/test_sigmoid.py]: 2025

**Review by SigureMo — DISMISSED** [review]

**Review by SigureMo — APPROVED** [review]

**Review by wanghuancoder — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]
